src/lib/eap_aka_sim/encode.c

```diff
@@ -845,28 +783,30 @@ static ssize_t encode_tlv_hdr(uint8_t *out, size_t outlen,
 	 *	length field in the buffer.
 	 */
 	total_len = ROUND_UP_POW2(len + 2, 4);
-	*p++ = da->attr & 0xff;			/* Type */
-	*p++ = total_len >> 2;			/* Length */
-	p += len;				/* Now increment */
+	FR_DBUFF_BYTES_IN_RETURN(&tl_dbuff, (uint8_t)da->attr, (uint8_t)(total_len >> 2));
 
-	FR_PROTO_HEX_DUMP(out, p - out, "Done TLV attribute");
+	FR_PROTO_HEX_DUMP(fr_dbuff_start(&work_dbuff), fr_dbuff_used(&work_dbuff), "Done TLV attribute");
 
-	return p - out;	/* AT_IV + AT_*(TLV) - Can't use total_len, doesn't include IV */
+	return fr_dbuff_set(dbuff, &work_dbuff);	/* AT_IV + AT_*(TLV) - Can't use total_len, doesn't include IV */
 }
 
 ssize_t fr_aka_sim_encode_pair(uint8_t *out, size_t outlen, fr_cursor_t *cursor, void *encoder_ctx)
+{
+	return encode_pair_dbuff(&FR_DBUFF_TMP(out, outlen), cursor, encoder_ctx);
+}
+
+static ssize_t encode_pair_dbuff(fr_dbuff_t *dbuff, fr_cursor_t *cursor, void *encoder_ctx)
 {
 	VALUE_PAIR const	*vp;
+	fr_dbuff_t		work_dbuff = FR_DBUFF_NO_ADVANCE(dbuff);
+	fr_dbuff_marker_t	m;
 	ssize_t			slen;
-	size_t			attr_len;
 
 	fr_da_stack_t		da_stack;
 	fr_dict_attr_t const	*da = NULL;
 	fr_aka_sim_encode_ctx_t	*packet_ctx = encoder_ctx;
 
-	if (!cursor || !out) return PAIR_ENCODE_FATAL_ERROR;
-
-	CHECK_FREESPACE(outlen, 4);		/* Attributes lengths are always multiples of 4 */
+	if (!cursor || !fr_dbuff_marker(&m, &work_dbuff)) return PAIR_ENCODE_FATAL_ERROR;
```

Comment:
Not sure why fr_dbuff_marker is included in this condition?

Reply:
Looking at it, no real need; in theory it might return NULL, but that's pretty paranoid. Hoisting it out of the `if`.